File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        
        m_dict = defaultdict(int)
        res = 0
        for i in nums:
            if  not  m_dict[i]:
                m_dict[i] = m_dict[i-1] + m_dict[i+1] +1
                m_dict[i-m_dict[i-1]] = m_dict[i]
                m_dict[i+m_dict[i+1]] = m_dict[i]
            res = max(res,m_dict[i])
        
        return res

File: Data Structures & Algorithms/two-integer-sum/submission-1.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {

        var res = new Dictionary<int,int>();

        for(int i =0; i < nums.Length ; i ++) {
            int diff = target - nums[i];
            if(res.ContainsKey(diff)) {
                return new int[]{res[diff],i};
            }
            res.Add(nums[i],i);
        }
        return new int[0];
    }
}
